test/UnitTest/Microsoft.AspNetCore.OData.Test/PublicApi/Microsoft.AspNetCore.OData.PublicApi.bsl

```diff
@@ -3359,15 +3412,18 @@ public class Microsoft.AspNet.OData.Formatter.Serialization.ODataResourceSetSeri
 public class Microsoft.AspNet.OData.Formatter.Serialization.ODataSerializerContext {
 	public ODataSerializerContext ()
 	public ODataSerializerContext (ResourceContext resource, Microsoft.OData.UriParser.SelectExpandClause selectExpandClause, Microsoft.OData.Edm.IEdmProperty edmProperty)
+	public ODataSerializerContext (ResourceContext resource, Microsoft.OData.UriParser.SelectExpandClause selectExpandClause, Microsoft.OData.Edm.IEdmProperty edmProperty, ODataQueryOptions queryOptions, Microsoft.OData.UriParser.ExpandedReferenceSelectItem expandedItem)
```

Comment:
Do we need a new overload, or can I call as ODataSerializerContext(resource, selectExpandClause, edmProperty){QueryOptions = queryOptions, ExpandSelectItem = expandSelectItem}?  Do we think the new pattern will be to *always* set these, and the current constructor is there just for backward compat? Would we mark the old one as deprecated?

Reply:
As we had discussed. This one will be the one that always gets called. Deciding against marking this as deprecated because the delta serializer still uses that. May be we can remove this when we refactor the Delta Serializer to share more code with Resource Serializer. 

---
In reply to: [287525311](https://github.com/OData/WebApi/pull/1724#discussion_r287525311) [](ancestors = 287525311)